File: python/algorithms/factorial_iterative.py
"""팩토리얼 (반복) 예시."""


def factorial_iterative(n: int) -> int:
    if n < 0:
        raise ValueError("n은 0 이상이어야 합니다.")

    result = 1
    for i in range(2, n + 1):
        result *= i
    return result


if __name__ == "__main__":
    print(factorial_iterative(5))  # 120

File: c/algorithms/factorial_iterative.c
#include <stdio.h>

long long factorial_iterative(int n){
    long long result=1;
    for(int i=2;i<=n;i++) result*=i;
    return result;
}

int main(void){
    printf("factorial_iterative(5) = %lld\n", factorial_iterative(5));
    return 0;
}
